### PR TITLE
Fix minor typos

### PR DESCRIPTION
**Pull request by marcwrobel**
Fix various minor typos in :

- project documentations,
- javadoc and comments,
- test class and method names,
- exception messages.

### CODE REVIEW

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng3023ReactorDependencyResolutionTest.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ public void testitMNG3023A() throws Exception
     /**
      * Test that reactor projects are included in dependency resolution.
      *
-     * I this pass, the dependency artifact should have the file $(basedir)/dependency/dependency-classes
+     * If this pass, the dependency artifact should have the file $(basedir)/dependency/dependency-classes
```
Comment (michael-o) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng3023ReactorDependencyResolutionTest.java:83]: passes?

File: core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/ConfigMojo.java [hunk 2]
```diff
@@ -118,7 +118,7 @@
     /**
      * A simple parameter of type {@link java.lang.Integer}.
      *
-     * @parameter property="config.intergerParam"
+     * @parameter property="config.integerParam"
```
Comment (michael-o) [inline core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/ConfigMojo.java:121]: Attention: Is this property used in the IT?

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng3023ReactorDependencyResolutionTest.java (hunk 1 above)
Comment (marcwrobel) [reply core-it-suite/src/test/java/org/apache/maven/it/MavenITmng3023ReactorDependencyResolutionTest.java:83]: Of course ! Fixed.

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng3023ReactorDependencyResolutionTest.java [hunk 3]
```diff
@@ -112,7 +112,7 @@ public void testitMNG3023B()
     /**
      * Test that reactor projects are included in dependency resolution.
      *
-     * I this pass, the dependency should have been installed, so the dependency artifact should have
+     * If this pass, the dependency should have been installed, so the dependency artifact should have
```
[the remark this answers is not in the document]
Comment (marcwrobel) [reply core-it-suite/src/test/java/org/apache/maven/it/MavenITmng3023ReactorDependencyResolutionTest.java:115]: Fixed.

File: core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/ConfigMojo.java (hunk 2 above)
Comment (marcwrobel) [reply core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/ConfigMojo.java:121]: It did not find any reference in this repo. And according to https://github.com/search?q=config.intergerParam&type=code there is no references in other repos either.

Do you think it's safer to roll back the change ?

Comment (michael-o) [reply core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/ConfigMojo.java:121]: Use grep, then we are sure.

Comment (marcwrobel) [reply core-it-support/core-it-plugins/maven-it-plugin-configuration/src/main/java/org/apache/maven/plugin/coreit/ConfigMojo.java:121]: No result when grepping on `intergerParam`.

**Pull request merged**